src/main/java/com/mraof/minestuck/entity/AnimatedPathfinderMob.java

```diff
@@ -0,0 +1,137 @@
+package com.mraof.minestuck.entity;
+
+import com.mraof.minestuck.entity.animation.MobAnimation;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.syncher.EntityDataAccessor;
+import net.minecraft.network.syncher.EntityDataSerializers;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.PathfinderMob;
+import net.minecraft.world.entity.ai.attributes.AttributeInstance;
+import net.minecraft.world.entity.ai.attributes.AttributeModifier;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.level.Level;
+
+import java.util.UUID;
+
+/**
+ * Abstract class for handling the actions of Geckolib animated mobs. It keeps track of which animation is in use via a MobAnimation and EntityDataAccessor
+ */
+public abstract class AnimatedPathfinderMob extends PathfinderMob
+{
+	/**
+	 * Retains the action from MobAnimation
+	 */
+	private static final EntityDataAccessor<Integer> CURRENT_ACTION = SynchedEntityData.defineId(AnimatedPathfinderMob.class, EntityDataSerializers.INT);
+	
+	private static final UUID FREEZE_MOB_UUID = UUID.fromString("e4c7543e-335c-4217-8d15-25c157e8ce88");
+	private static final AttributeModifier STATIONARY_MOB_MODIFIER = new AttributeModifier(FREEZE_MOB_UUID, "Stationary Animation Modifier", -10, AttributeModifier.Operation.MULTIPLY_TOTAL);
```

Comment:
How about just -1? That'd change the speed to 0 rather than negative, and thus play nicer with other attribute modifiers.